scripts/convert_config.py

```diff
@@ -0,0 +1,460 @@
+             }
+         },
+         "another_component": {
+             "an_option": {
+                 "section": "some_section",
+                 "option" : "some_old_option_name",
+                 "default": "default",
+             },
+             "another_option": {
+                 "section": "another_section",
+                 "option" : "another_old_option_name",
+                 "default": "default_value",
+             }
+         }
+     }
+ }
+
+ Check "scripts/templates/2.0/mapping.json" file for an example.
+
+ The idea is to provide new templates, mapping, and adjust script for new
+ versions of the configuration, allowing the user to easily convert from a
+ version of the configuration file to the next.
+"""
+
+COMPONENTS = ["agent", "verifier", "tenant", "registrar", "ca", "logging"]
+
+# Config files can be merged together, reading from the system to the
+# user.
+CONFIG_FILES = ["/usr/etc/keylime.conf", "/etc/keylime.conf",
+                os.path.expanduser("~/.config/keylime.conf")]
+
+if "KEYLIME_CONFIG" in os.environ:
+    CONFIG_FILES.insert(0, os.environ["KEYLIME_CONFIG"])
+
+
+def get_config(path) -> configparser.RawConfigParser:
+    """
+    Read configuration files and merge them together
+    """
+
+    if path:
+        if not os.path.exists(path):
+            raise Exception(f"Path {path} does not exist")
+
+        if not os.path.isfile(path):
+            raise Exception(f"Path {path} is not a file")
+
+        config_files = [path]
+    else:
+        print(f"No input provided: using {CONFIG_FILES} as input")
+        config_files = CONFIG_FILES
+
+    config = configparser.RawConfigParser()
+
+    if not any(os.path.exists(c) for c in config_files):
+        print(
+            f"Config file not found in {config_files}. Please set "
+            f"environment variable KEYLIME_CONFIG or see {__file__} "
+            "for more details."
+        )
+    else:
+        # Validate that at least one config file is present
+        config_files = config.read(config_files)
+        print(f"Reading configuration from {config_files}")
+
+    return config
+
+
+def output_component(component, config, template, output):
+    """
+    Output the configuration file for a component
+    """
+
+    print(f"Writing {component} configuration to {output}")
+
+    with open(template, "r") as tf:
+        t = tf.read()
+
+        j2 = Template(t)
+
+        r = j2.render(config)
+
+        with open(output, "w") as o:
+            print(r, file=o)
+
+
+def output(args, config, templates):
+    """
+    Output the requested files
+    """
+
+    flat = [s for ss in args.component for s in ss]
+
+    if not flat:
+        # If no component was provided, use all available
+        components = COMPONENTS
+    else:
+        clean = list(set(flat))
+        for c in clean:
+            if not c in COMPONENTS:
+                print(f"Unknown component {c}, skipping")
+                clean.remove(c)
```

Comment:
Ahh Rust would complain here!

```python
>> a = [1, 2, 3, 4]
>>> for i in a:
...   print(i)
...   a.remove(i)
... 
1
3
```

Reply:
Thank you for pointing this, I'll change the code.